Fix/usrmerge sbin

smoser opened the pull request:
These are change to packages that ship files in /sbin, moving them to /usr/bin/

smoser committed:
apk-tools - usrmerge-sbin

smoser committed:
e2fsprogs - usrmerge-sbin

smoser committed:
cifs-utils - usrmerge-sbin

smoser committed:
cilium - usrmerge-sbin

smoser committed:
cryptsetup - usrmerge-sbin

smoser committed:
dhclient - usrmerge-sbin

smoser committed:
efs-utils - usrmerge-sbin

smoser committed:
fuse2 - usrmerge-sbin

smoser committed:
iptables - usrmerge-sbin

smoser committed:
iproute2 - usrmerge-sbin

smoser committed:
iptables - usrmerge-sbin

smoser committed:
keyutils - usrmerge-sbin

smoser committed:
lvm2 - usrmerge-sbin

smoser committed:
nfs-utils - usrmerge-sbin

smoser committed:
openrc - usrmerge-sbin

smoser committed:
runit - usrmerge-sbin

smoser committed:
su-exec - usrmerge-sbin

smoser committed:
shadow - usrmerge-sbin

smoser committed:
systemd - usrmerge-sbin

smoser committed:
tini - usrmerge-sbin

smoser committed:
utmps - usrmerge-sbin

justinvreeland commented on dhclient.yaml:
Epoch

justinvreeland commented on fuse3.yaml:
epoch

xnox reviewed (changes requested):
Runtime depends on "merged-sbin" virtual package please.

xnox commented on e2fsprogs.yaml:
```diff
@@ -41,6 +41,8 @@ pipeline:
   - uses: autoconf/make
 
+  - uses: usrmerge/sbin
+
```
@smoser drop this?

a-crate commented:
Missing `glibc`, `net-tools`, and `util-linux` merges?

justinvreeland commented on iptables.yaml:
```diff
@@ -42,7 +42,7 @@ pipeline:
         --prefix=/usr \
         --mandir=/usr/share/man \
-        --sbindir=/sbin \
+        --sbindir=/usr/bin \
```
When I install this iptables package cilium builds fail because It's looking for `/usr/sbin/` iptables. I believe our plan is to symlink `/sbin` to `/usr/bin` not `/usr/sbin` so this will work. But I am curious how I test this currently.  Do I need to use the base-layout package from: https://github.com/wolfi-dev/os/pull/42522/ ?

roaksoax commented on tini.yaml:
```diff
@@ -30,21 +30,14 @@ pipeline:
+      - runs: install -Dm755 build/tini-static ${{targets.subpkgdir}}/usr/bin/tini-static
     description: Static build of tini
 
-  - name: tini-compat
```
@justinvreeland @a-crate let's not forget to make this one an empty package that we can clean-up later.

a-crate replied:
Done.

a-crate replied:
The change providing tini-compat got reverted at some point, this still needs to be added.

smoser commented on net-tools.yaml:
```diff
@@ -79,17 +79,6 @@ pipeline:
-      for cmd in arp ipmaddr iptunnel nameif plipconfig rarp slattach; do
-        ln -s "../usr/bin/${cmd}" "${{targets.destdir}}/sbin/${cmd}"
-      done
-
```
I just dropped them both because the symlinks are inconsistent with everything else (and were new in the most recent epoch).

smoser committed:
net-tools - usrmerge-sbin

smoser committed:
zfs - usrmerge-sbin

smoser commented:
> Missing `glibc`, `net-tools`, and `util-linux` merges?

added net-tools. still missing glibc and util-linux here.

justinvreeland commented:
> > Missing `glibc`, `net-tools`, and `util-linux` merges?
> 
> added net-tools. still missing glibc and util-linux here.

I have a PR for glibc I'll add here

smoser committed:
util-linux - usrmerge-sbin

smoser committed:
efs-utils - [MANUAL] comment changes so yam/ruamel does not mangle

smoser commented:
this is the script i'm using to add all the machinable changes. https://gist.github.com/smoser/7da4ed38fb06e80d199ce28e07c20d1b#file-sbin-mergy

Commit:
wofli-baselayout: merged-sbin baselayout package

Commit:
clilium-1.16: fixup sbin path

Commit:
utmp-docs: Fix tests

Commit:
e2gsprogs: update configure options

Commit:
iptables: remove broken tests

https://github.com/wolfi-dev/os/pull/42559 did run any of the tests on
the PR so broken tests made it through this only reverts iptables.

sergiodj commented on apk-tools.yaml:
```diff
@@ -1,13 +1,14 @@
-  epoch: 2
-  description: "apk-tools (Wolfi package manager)"
+  epoch: 3
+  description: apk-tools (Wolfi package manager)
```
You'll want to do `yml.preserve_quotes = True` (on https://gist.github.com/smoser/7da4ed38fb06e80d199ce28e07c20d1b) in order to avoid these bogus changes.

xnox commented on apk-tools.yaml:
```diff
@@ -1,13 +1,16 @@
       - ca-certificates-bundle
+      - merged-sbin
+      - merged-sbin
+      - wolfi-baselayout
```
duplicate runtime dep

justinvreeland replied:
I honestly thought yam took care of duplicate deps.

xnox commented on cifs-utils.yaml:
```diff
@@ -1,10 +1,15 @@
+    runtime:
+      - merged-sbin
+      - merged-sbin
+      - wolfi-baselayout
```
duplicate runtime dep

xnox commented on cryptsetup.yaml:
```diff
@@ -2,15 +2,20 @@
+    runtime:
+      - merged-sbin
+      - merged-sbin
+      - wolfi-baselayout
```
duplicate runtime dep

xnox commented on dhclient.yaml:
```diff
@@ -1,14 +1,17 @@
       - iproute2
+      - merged-sbin
+      - merged-sbin
+      - wolfi-baselayout
```
duplicate runtime dep

a-crate commented on paxmark.yaml:
```diff
@@ -26,7 +28,7 @@ pipeline:
   - runs: |
       cargo auditable build --release
-      install -Dm755 target/release/paxmark "${{targets.destdir}}"/usr/sbin/paxmark
+      install -Dm755 target/release/paxmark "${{targets.destdir}}"/usr/bin/paxmark
```
@zmarano This is /usr/sbin, not /sbin

zmarano replied:
Yeah I ust found I put this in the wrong one. Moved it to your branch.

Commit:
Run script to add new deps and EPOCH bump

Commit:
glibc: fixup sbin path

justinvreeland commented:
Ohhhh, the other ones finished wayyyy quicker I wonder if the additional dependencies are causing problems.   We can pull the list of specific packages and subpackages from the spreadsheet if that's the case : (

zmarano committed:
busybox - usrmerge 1 sbin

zmarano commented on iptables.yaml:
```diff
@@ -77,22 +81,23 @@ subpackages:
+          setcap cap_net_raw,cap_net_admin+eip ${{targets.contextdir}}/usr/bin/xtables-legacy-multi
+          setcap cap_net_raw,cap_net_admin+eip ${{targets.contextdir}}/usr/bin/xtables-nft-multi
     test:
-      environment:
```
Need to restore this test environment. (lines 84-87)

zmarano replied:
Fixed in b4289b3

zmarano committed:
fix iptables test

zmarano committed:
Remove problematic getcap test from iptables.

xnox commented:
Did rebase to resolve conflicts in openrc and dhclient.

xnox commented:
Please force-merge, the pod termination on glibc is a flake.
